Sources/Datadog/FeaturesIntegration/WebView/WKUserContentController+Datadog.swift

```diff
@@ -16,10 +16,24 @@ public extension WKUserContentController {
     internal func __addDatadogMessageHandler(allowedWebViewHosts: Set<String>, hostsSanitizer: HostsSanitizing) {
         let bridgeName = DatadogMessageHandler.name
 
+        let contextProvider = (Global.rum as? RUMMonitor)?.contextProvider
+        let logEventConsumer = WebLogEventConsumer(
+            userLogsWriter: LoggingFeature.instance?.storage.writer,
+            internalLogsWriter: InternalMonitoringFeature.instance?.logsStorage.writer,
+            dateCorrector: LoggingFeature.instance?.dateCorrector,
+            rumContextProvider: contextProvider
+        )
+        let rumEventConsumer = WebRUMEventConsumer(
+            dataWriter: RUMFeature.instance?.storage.writer,
+            dateCorrector: RUMFeature.instance?.dateCorrector,
+            webRUMEventMapper: WebRUMEventMapper(),
+            contextProvider: contextProvider
+        )
```

Comment:
Same here, all these components must be injected, so we can mock it in tests. Also, there are no tests for `WKUserContentController+Datadog` that would guarantee its all consistent.

Reply:
what do you mean by consistent?
as discussed before (maybe in Slack though) i couldn't find a way to unit test the behavior of this component, so i can't see where we can make use of mocks.

Reply:
By consistency I mean all the things that may go wrong either by regression on our end or by wrong configuration from the user. Some examples:
* misconfiguration:
   * enabling this integration while RUM and Logging features are disabled
   * enabling this while no `Global.rum` is registered

* regression:
   * given Logging/RUM enabled, when receiving event from Browser SDK, it gets passed to the right writer;

Reply:
- enabling this integration while RUM and Logging features are disabled
- enabling this while no Global.rum is registered

these can be legit use cases, i don't know how to handle these cases. wdyt @mariusc83 ?